docs/docsite/requirements.in

```diff
@@ -0,0 +1,10 @@
+# This requirements file is used for AWX latest doc builds.
+
+# Note that this set of requirements inherits pinned versions from a previous RST source repository.
```

Comment:
Semantically, this should be a loose list of deps for the docs venv but as it is not, it's more of a set of workarounds. Because, there's pins and upper bounds but no lower bounds. And maybe, not all of these are even direct deps.

Reply:
@oraNod you didn't add the lower bounds, you introduced the upper bounds which is the opposite of what's expected...

Reply:
@webknjaz Bah. I was afraid I would mess that up. I still get tangled up with dependencies. Thanks for catching.

Before I add another change I'd like to ask.

``>=minimum_version`` this sets the lower bound right?

However what I think we actually need is an upper bound. e.g. the Jinja version should not be greater than 3.1. So what I think we need is actually something like this ``Jinja>=x,<3.1`` but I really don't know what ``x`` would be here.

Again I want to clean up these requirements and have things in proper order. In the previous repository (private `product-docs`) the requirements have been carried over from different versions without a lot of maintenance and I think purpose has been somewhat lost. I also don't know a lot of the history there.

While I appreciate this feedback and enjoy what I'm learning here I wonder if maybe we shouldn't get too caught up on the bounds and pins because we want to remove those eventually. What do you think?

Reply:
Yep, `>= version` is the right way. I also like having that whitespace for visual separation. I think, it's more readable.

If you don't know the lowest versions, don't add any limits at all.

How about having things with no boundaries here? Then, when running pip-compile, supply those extra restrictions via CLI args (I think, that's `-P`). This way, you'll have constraints that correspond the current state of the repo but the input file will be ready for further pin upgrades that you could do through follow-up PRs, one by one.
If one of those bumps causes problems, you'll see the reason for the original restrictions and maybe, reintroduce the incoming constraints file or just fix those problems together with the pin updates, as an atomic change.

Reply:
Top class suggestion @webknjaz :tada: 

I've moved the pins to the pip-tools configuration file. That feels like a big improvement. How about if we just keep the pins there exactly as they were from the previous repo? Not sure if you'd feel strongly about changing the pins there.

Reply:
Not sure I understand the question.. Also, I didn't realize you could even put those into the config file. All I was suggesting is that you pass the args to the CLI command you executed locally and not save those pins. Neither in the input file, nor the config. They'll already be in the constraint/pins/lockfile if you run the command like that. And in follow-ups, you'd again just upgrade the packages one by one, removing the limitations in your local cli command invocation. Thst would generate new constraint updates which would be the only things in the follow-up PRs, and with no input/config changes necessary, if they are not checked in.

Reply:
But I suppose it's good enough as it is, if you'll work on removing them. The pip-tools config is global, so it may end up being used in other contexts, like making constraints for the test envs or, development, or production runtimes — all the things unrelated to the docs.